src/Foundation/ArrayHelper.php

```diff
@@ -17,15 +17,13 @@ final class ArrayHelper
      * To distinguish the two it evaluates keys of PHP array:
      * - if there are only numeric keys (0...N) then it returns true
      * - otherwise, if there are string keys it returns false
-     *
-     * @param mixed[] $arr
      */
     public static function isAssoc(array $arr) : bool
     {
         if ($arr === []) {
             return false;
         }
 
-        return array_keys($arr) === range(0, count($arr) - 1);
+        return array_keys($arr) !== range(0, count($arr) - 1);
```

Comment:
This one was interesting and worked because there was 
```
!(is_array($data) && ArrayHelper::isAssoc($data))
``` 

instead of 

```
!is_array($data) || ArrayHelper::isAssoc($data)
```

Reply:
Makes sense!